blog/docs/articles/workflow-as-function-flow.md

```diff
@@ -0,0 +1,329 @@
+
+- be single activity
+- include other activities to control logic
+- be combined with other (executing) activities to form so called continuation
+
+Functions are named automatically based on workflow or activity names or explicitly by user using custom attributes.
+
+Each function will then become a dedicated entry point that will
+
+- have input that is built based on workflow data
+- can produce one or more outputs
+- will have Knative trigger associated with the function
+
+![Slice workflow into functions](/blog/articles/images/workflow-function-flow-function-slice.png)
+*Slice workflow into functions*
+
+### Inputs and outputs as Cloud Events
+
+Once the functions are derived from the workflow, they will always have one event as input
+and can produce one or more events as outputs. The logic defined in the workflow and actual data context of a particular workflow instance will drive the number of events produced. Regardless of the event being an input or output it is
+
+- built based on data objects defined in the workflow
+- filed automatically based on execution context
+- uses **type** attribute as link to function
+- uses **source** attribute as identifier of the function being executed including unique identifier of workflow instance for correlation purposes
+
+![Inputs and outputs as events](/blog/articles/images/workflow-function-flow-function-io.png)
+*Inputs and outputs as events*
+
+Functions always exchange events through Knative Eventing broker. They never call each other directly to make sure they are completely decoupled and event driven. This will allow for greater scalability as each function call can be handled by another replica of the service.
+
+### Deployment
+
+First of all, the heavy work that requires parsing workflow definition, slicing that into
+functions and events and packaging into deployable unit is done at build time.
+
+Automatiko project that implements this concept is a java based implementation that takes advantage of [GraalVM](https://www.graalvm.org) to be able to compile into native executable that brings small memory footprint at runtime and lightning fast startup time.
+
+With that said, workflow as a function flow with Automatiko compiles into single service and by that into a single container image, that includes all the functions. Each and every function can be invoked at any time on any replica of the service.
+
+At the same time (during build) Knative manifest file is generated to ease deployment to the cluster. That consists of
+
+- Knative serving service
+- sink binding to inject broker location
+- triggers for each function created out of the workflow
+
+So this serves as a starting point to have fully runnable service directly after build. It can be modified further to adopt to particular configuration of Knative in the cluster.
+
+Enough theory, let's look at example of this concept.
```

Comment:
```suggestion
The heavy work that requires parsing workflow definition, slicing that into functions and events, and packaging this in to a deployable unit, is done at build time.

The Automatiko project implements this concept. It is a java based implementation that takes advantage of [GraalVM](https://www.graalvm.org) to compile into a native executable, which has a small memory footprint at runtime and lightning fast startup time.

Workflow as a function flow with Automatiko compiles into single service and by that into a single container image, that includes all the functions. Each and every function can be invoked at any time on any replica of the service.

At the same time during build, a Knative manifest file is generated to ease deployment to the cluster. The Knative manifest file consists of:

- a Knative serving service
- a sink binding to inject broker location
- triggers for each function created out of the workflow

This serves as a starting point to have fully runnable service directly after build. It can be modified further to adopt to particular configuration of Knative in the cluster.
```